docs/rules/no-reassign-constant.md

```diff
@@ -0,0 +1,46 @@
+# Disallow Reassignment of Constants (no-reassign-constant)
+
+Variables that are declared as constants but are later reassigned, are no longer constant, and should be declared with let or var.
```

Comment:
This introduction needs to be longer, a few sentences at least. You should explain that reassigning to constants throws an error in ES6 in both strict and nonstrict mode and show an example.